src/systems/filecoin_mining/storage_mining/storage_miner_actor.go

```diff
@@ -72,71 +97,76 @@ func (sm *StorageMinerActor_I) submitFaultReport(
 		NewTerminatedFaults_: newTerminatedFaults,
 	}
 
-	sm.SectorTable().Impl().TerminationFaultCount_ = util.UVarint(0)
-
 	panic(faultReport)
 	// TODO: Send(SPA, ProcessFaultReport(faultReport))
+
+	h, st := a.State(rt)
+	st.SectorTable().Impl().TerminationFaultCount_ = util.UVarint(0)
```

Comment:
perhaps `st.AddTerminationCount(util.UVarint(0))` ?

Reply:
(keep the actor _code_ as algorithm-oriented as we can?)

Reply:
Makes sense (but should this be `SetTerminationCount` rather than `AddTerminationCount`?)

Reply:
dont think so, i think when we notice termination counts, we want to increment a count?